src/mappings/metadata.ts

```diff
@@ -0,0 +1,39 @@
+// V2 Pool Factory event handling and mapping
+
+import { log, json, Bytes } from '@graphprotocol/graph-ts'
+import { Metadata } from '../../generated/PoolMetadata/PoolMetadata'
+import { Pool } from '../../generated/schema'
+
+
+export function handleMetadata(event: Metadata): void {
+  // parse
+  let res = json.try_fromBytes(Bytes.fromUTF8(event.params.data) as Bytes);
+  if (res.isError) {
+    log.info('unable to parse metadata for pool: {}', [event.params.pool.toHexString()]);
+    return;
+  }
+  let data = res.value.toObject();
+
+  // update pool object
+  let pool = Pool.load(event.params.pool.toHexString());
+  pool.name = '';
+  pool.description = '';
+  pool.website = '';
```

Comment:
Whats the reasoning behind initializing these as empty strings vs just not setting it if the field doesn't exist?

Reply:
This is just to make sure the `clear` method works to remove all pool metadata